### DISCUSSION TITLE
VPC REST API

### DISCUSSION THREAD

**Discussion by samueldion** [AxisCommunications/acap-integration-examples-aws#21] [General] [answered]
Hello all,

Just a general question here.  Can the API be created and functional just as well using a REST API within a VPC.  What would be the easiest way to go about it to make that edit in AWS?  I have everything working with the HTTP API and want to make the switch.
Note that I am novice in anything cloud architecture 😊

Thank you,
Samuel

**Comment by mattias-kindborg-at-work** [comment]
Hi @samueldion!

To frame the question, are we talking about [sending images to S3](https://github.com/AxisCommunications/acap-integration-examples-aws/tree/main/images-to-aws-s3) or [telemetry to AWS IoT Core](https://github.com/AxisCommunications/acap-integration-examples-aws/tree/main/telemetry-to-aws-iot-core)?

**Reply by samueldion** [reply]
Hi Mattias,
Sending images to S3!

**Reply by mattias-kindborg-at-work** [reply]
Images are sent from the Axis camera to the [API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/welcome.html). A [REST API](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-create-api.html) is just a set of REST resources on the API Gateway. I haven't tested, but I can't think of any issues with setting up the REST resources, given you pass the important parts of the request to the Lambda function (like the filename specified as a request header), and given that the API Gateway can be accessed from a camera in the VPC.

**Reply by samueldion** [reply]
Hello @mattias-kindborg-at-work ,
What is the content body that is sent from the camera?  I am trying to implement it in Postman and get it working as a REST API, but I am just getting an issue by not knowing how the Axis camera actually send the HTTPS POST request.  I don't know if my question make sense?  Anyway, just wondering how to get it to work in Postman :)

**Reply by mattias-kindborg-at-work** [reply, marked as answer]
You can use the following properties in Postman to simulate an Axis camera sending an image:

- Method: `POST`
- Path: `<url>`
- Query parameters
  - `accessToken`: `<access token>`
- Headers
  - `Content-Disposition`: `attachment; filename="<filename>"`
- Body
  - Binary: _Browse and select file_

where `<url>` is the recipient, `<access token>` is the access token and `filename` is the name of the image you are in the process of sending. You can also [follow along in the code](https://github.com/AxisCommunications/acap-integration-examples-aws/blob/main/images-to-aws-s3/src/upload-to-s3/index.js) to see what the receiving function expects from the request.

**Reply by samueldion** [reply]
Thank you Mattias, I indeed followed the code and it worked well. 
Cheers,

**Reply by samueldion** [reply]
Hey Mattias,
I am coming back to this because I need to add a custom header to the Axis call so that it can reach the proper API in the VPC.  I just don't know where to put it in the Axis settings.  I am not seeing the option to do so in the Camera interface...
Thank you,
Samuel

**Reply by mattias-kindborg-at-work** [reply]
Hi Samuel! Unfortunately you can't specify a custom header in the camera UI 😞
